Add Selected CaloJet Collection to MiniAOD

jingyuluo opened the pull request:
This PR adds the selected calojet collection to MiniAOD, currently stored bits are P4, EMEnergyFraction, HadronicEnergyFraction, towersArea (need to further check with JetMET group). The jet selection is pt>20GeV.

jingyuluo committed:
add CaloJets to miniAOD

jingyuluo committed:
set the vertex coordinates to zeros, add jet selection pT>10GeV

jingyuluo committed:
change jet cut to pt>20GeV

cmsbuild commented:
A new Pull Request was created by  @jingyuluo  for master.

It involves the following packages:

PhysicsTools/PatAlgos


@perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please review it and eventually sign? Thanks.
@TaiSakuma, @gouskos, @imarches, @ahinzmann, @acaudron, @mmarionncern, @rappoccio, @jdolen, @nhanvtran, @gpetruc, @gkasieczka, @schoef, @ferencek, @mverzett, @mariadalfonso, @pvmulder, @seemasharmafnal, @JyothsnaKomaragiri this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -189,7 +189,7 @@ def miniAOD_customizeCommon(process):
 	 lazyParser = cms.bool(True) )
     task.add(process.caloJetMap)
-    process.patJets.userData.userFloats.src += [ cms.InputTag("caloJetMap:pt"), cms.InputTag("caloJetMap:emEnergyFraction") ]
+#    process.patJets.userData.userFloats.src += [ cms.InputTag("caloJetMap:pt"), cms.InputTag("caloJetMap:emEnergyFraction") ]
```
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

jingyuluo replied:
Hi@slava77, the pt and emEnergyFraction of caloJetMap are not needed when the calo jet is implemented. Should I just remove this line from the code?

slava77 replied:
I think that it is still needed:
- the threshold in slimmed calojets is not necessarily the same
- it would take an extra effort to loop over the slimmed calojets to get a match and compute these userfloats

gpetruc replied:
Hi Slava,
the user floats were only supposed to be a temporary thing for early data commissioning, and I believe nobody is using them in production.
I think it's best to remove them now from the userfloats, and if there is a usecase for going below 20 GeV of raw calo jet pt then we change the configuration of the calo jet slimmer accordingly.

slava77 replied:
It looks like my comment from a week ago still applies:https://github.com/cms-sw/cmssw/pull/19446/files#r124269867
Based on the discussion, it looks like it is OK to drop the userFloats.

So, the commented out line should be removed or there should be a clear comment added why it is still needed

slava77 replied:
@jingyuluo 
please clarify when I should expect an update here.
Thank you.

jingyuluo replied:
Hi @slava77 , sorry... I'll remove this line.
Thanks!

slava77 commented on PhysicsTools/PatAlgos/plugins/SlimCaloJetSelector.h:
```diff
@@ -0,0 +1,84 @@
+        produces< reco::CaloJetCollection> ("selectedCaloJets");
+
+
+    }
```
there are a few places with empty lines for no particular reason

jingyuluo replied:
Hi @slava77 , I'll remove those empty lines.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

rappoccio commented:
Just for my own information, why is it better to add the CaloJets here instead of, say, PFClusterJets instead?

cmsbuild commented:
+1
Tested at: 0c1d89025f38751d30043531b31934f36dc94560
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/20944/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3b01ccc4e3b8104d08bcfbecec39c470c02890cc
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/20944/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/20944/git-merge-result

cmsbuild commented:
Comparison job queued.

jingyuluo commented:
Hi @rappoccio , PF jet is tuned for prompt jet, so it can be biased for displaced jet search. Also, using Calojets can make our analysis more inclusive to different displaced signatures.

rappoccio commented:
@jingyuluo not PF Jets, PF CLUSTER jets. They use the PF Clusters without tracks attached to them, and are closer to calo jets. They, however, have the same clustering as PF jets, so your association will be much more sensible. Otherwise the calo towers that make the calo jets don't necessarily match with the jets in miniaod.

slava77 commented on PhysicsTools/PatAlgos/plugins/SlimCaloJetSelector.h:
```diff
@@ -0,0 +1,84 @@
+
+#include <vector>
+
+class SlimCaloJetSelector : public edm::stream::EDFilter<> {
```
For the framework scheduling it would be better to have this module to be an EDProducer.
Also, to be in line with other slimmers, the name should be something like PATCaloJetSlimmer

monttj replied:
Not only naming convention but also contents is not in line with other slimmers.
Output is not really PAT object but RECO object. So as far as I understand, it can not be simply PATCaloJetSlimmer. 

Taejeong

gpetruc replied:
Ok, then maybe it can be called just `CaloJetSlimmer`?

slava77 replied:
this makes sense if it will never ever include any dependence on PAT classes.
Would it or would it not?

jingyuluo replied:
Hi, maybe we can call it CaloJetSlimmer as @gpetruc suggested. We only need the basic information (p4 and JEC) from CaloJets for our analysis, so I think PAT would not be needed. (at least for displaced jet analysis)

slava77 commented on PhysicsTools/PatAlgos/plugins/SlimCaloJetSelector.h:
```diff
@@ -0,0 +1,84 @@
+    filter_( params.exists("filter") ? params.getParameter<bool>("filter"): false),
+    selector_( cut_ )
+    {
+        produces< reco::CaloJetCollection> ("selectedCaloJets");
```
instance value seems to be unnecessary, just ``` produces< reco::CaloJetCollection> () ``` is enough

slava77 commented on PhysicsTools/PatAlgos/plugins/SlimCaloJetSelector.h:
```diff
@@ -0,0 +1,84 @@
+    virtual ~SlimCaloJetSelector() {}
+
+    virtual void beginJob() {}
+    virtual void endJob() {}
```
no need for empty begin/endJob. Please remove.

slava77 commented on PhysicsTools/PatAlgos/plugins/SlimCaloJetSelector.h:
```diff
@@ -0,0 +1,84 @@
+
+        for (edm::View<reco::CaloJet>::const_iterator ibegin = h_jets->begin(),
+                iend = h_jets->end(), ijet = ibegin;
+                ijet != iend; ++ijet ){
```
```for (auto const& jet : *h_jets){``` 
is shorter and more effective

slava77 commented on PhysicsTools/PatAlgos/python/slimming/MicroEventContent_cff.py:
```diff
@@ -8,6 +8,7 @@
         'keep *_slimmedMuons_*_*',
         'keep *_slimmedTaus_*_*',
         'keep *_slimmedTausBoosted_*_*',
+        'keep *_selectedCaloJets_*_*',
```
to follow the naming convention, please call this slimmedCaloJets
("selected" implies just a selection, not modification that zeroes out values as a part of slimming)

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/20944/summary.html

Comparison Summary:
 - You potentially added 3 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1754229
 - DQMHistoTests: Total failures: 14573
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1739490
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

monttj commented:
@jingyuluo 

I think the output name should be "slimmedCaloJets" if it is only we can get the calo jet
and I believe it can be done by following PAT configuration sequence so that it can benefit from all PAT features. 

Regards,
Taejeong

jingyuluo committed:
implement slava's comments, store the jetArea in CaloJet for JEC

cmsbuild commented:
Pull request #19446 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

jingyuluo commented:
Hi, I implemented slava and Taejeong's comments (changed the name convention, switch to EDProducer, etc.) and stored the jetArea in CaloJet for JEC.

jingyuluo committed:
small fix

cmsbuild commented:
Pull request #19446 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 295a00e42d1876c53c40e635fd5c26bf5004c8e6
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21011/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/1974578bdc7b4ef3a501f360274c7bec46ac7066
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21011/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21011/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21011/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins-workarea/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-19446/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17
   * /build/cmsbld/jenkins-workarea/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-19446/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19

Comparison Summary:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1756063
 - DQMHistoTests: Total failures: 135
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1755762
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented:
@cmsbuild please test workflow 136.7611,1325.5

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented:
@cmsbuild please abort
 I forgot that the reminiaod wfs are still broken

cmsbuild commented:
Jenkins tests are aborted.

slava77 commented on PhysicsTools/PatAlgos/plugins/CaloJetSlimmer.cc:
```diff
@@ -0,0 +1,63 @@
+    cut_( params.getParameter<std::string>("cut") ),
+    selector_( cut_ )
+    {
+        produces< reco::CaloJetCollection> ();//("slimmedCaloJets");
```
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant.

jingyuluo replied:
Hi @slava77 ,it's not needed here, I'll remove it.

jingyuluo committed:
remove the commented out part

cmsbuild commented:
Pull request #19446 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented on PhysicsTools/PatAlgos/plugins/CaloJetSlimmer.cc:
```diff
@@ -0,0 +1,63 @@
+        for (auto const& ijet : *h_jets){
+           if( selector_(ijet) ){
+              reco::CaloJet::Specific tmp_specific;
+              const reco::Candidate::Point orivtx(0,0,0);
```
CaloJets in AOD have the PV assigned to the underlying reco::Candidate.
I did not find explicit dependence on vz, but my checks were really spotty.
Is it intentional to set this to 0 here (e.g. to save space, just because the PV can be reassigned later in analysis)?

jingyuluo replied:
Hi @slava77, yes, the vertex coordinates are set to 0 in order to save space. We can set the the vertices later in the analysis. (Actually, in the displaced jet analysis, the calo jets should be assigned to secondary vertices instead of the PV).

slava77 commented:
@cmsbuild please test workflow 136.7611,1325.5

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 3fad7622456e971da940bb8d671f050569fbfdda
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21146/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/e4d2a498625472141e9db6dfe617f5782f469833
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21146/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21146/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21146/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1760883
 - DQMHistoTests: Total failures: 15076
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1745641
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

jingyuluo committed:
remove the commented out line for pt and emEnergyFraction in CaloJetMap

cmsbuild commented:
Pull request #19446 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 4f40e3ab1bd83231ccd566117458ab269cf6af9e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21181/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/f19409072807f1553ebbe35357dabadf35f91d3e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21181/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21181/git-merge-result

cmsbuild commented:
Comparison job queued.

slava77 commented:
Looking at comparisons, there are now empty plots in miniAOD DQM corresponding to removal of the calojet information
e.g. in wf 4.53
<img width="836" alt="wf4 53_pt_calojet" src="https://user-images.githubusercontent.com/4676718/27875055-bfd62a34-6166-11e7-85c5-9660eadd1c6e.png">

@gpetruc commented earlier in https://github.com/cms-sw/cmssw/pull/19446#discussion_r124480312
> the user floats were only supposed to be a temporary thing for early data commissioning, and I believe nobody is using them in production

Clearly there is a use case (perhaps abandoned or irrelevant?)
does anybody rely on these?
@ahinzmann @rappoccio @jdolen 
@dmitrijus

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21181/summary.html

Comparison Summary:
 - You potentially added 2 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1786918
 - DQMHistoTests: Total failures: 44332
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1742420
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented:
@ahinzmann @rappoccio @jdolen @knash @schoef @seemasharmafnal 
please check if empty plots mentioned in https://github.com/cms-sw/cmssw/pull/19446#issuecomment-313160869
are important. They are empty due to drop of calojet userFloats in the slimmed PF jets.

If they are important, this PR will need some updates to recover the plots: the desired change is apparently to use the slimmedCaloJets directly; but I can imagine that it's easier to restore the calojet userFloats now and turn them off later once DQM catches up (again, if these plots are needed for validation rather than just being obsolete).

ahinzmann commented:
I'd say these plot at rather important for validation while the new pixel detector is being commissioned this year (and calojets are a reference). Let's better not break them.

slava77 commented:
@jingyuluo 
it seems most practical to undo commits in PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py  (git rebase -i to drop the commits, if you are familiar with this)

The follow up actions would be to have the DQM migrate from these userFloats to slimmedCaloJets and then drop the userFloats
@gpetruc @arizzi  please confirm that this is OK

jingyuluo commented:
Hi @slava77 , I'm not very familiar with git rebase (and there is no commit that only involves miniAOD_tools.py...), would it be okay to just add the line back and make a new commit?

slava77 commented:
On 7/6/17 9:02 AM, jingyuluo wrote:
> Hi @slava77 <https://github.com/slava77> , I'm not very familiar with
> git rebase (and there is no commit that only involves
> miniAOD_tools.py...), would it be okay to just add the line back and
> make a new commit?

yes, it should be OK

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19446#issuecomment-313440860>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbrqEexM19YPaPi69n2lnw_-7-_0tks5sLQT8gaJpZM4OGnTG>.
>

slava77 commented:
@jingyuluo 
is the update going to still happen today?
Please let me know.

jingyuluo commented:
Hi @slava77 , sorry for the delay, I'm also working on another PR... I'll update this one now

jingyuluo committed:
recover the userfloat for pt, emEnergyFraction of caloJetMap

cmsbuild commented:
Pull request #19446 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

@jingyuluo 
thank you for the quick update

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a933638f1bd3870b903b59318a273d00374a8722
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21231/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/6625d2da7769faa8b12de4b3703335f117f040fb
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21231/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21231/git-merge-result

cmsbuild commented:
Comparison job queued.

slava77 commented:
@cmsbuild please test

It looks like the last comparison job got stuck for some reason at result upload

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21231/summary.html

Comparison Summary:
 - You potentially added 5 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1790655
 - DQMHistoTests: Total failures: 29358
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1761131
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/19446	a933638
- added slimmedCaloJets to miniAOD
- jenkins tests pass and comparisons with baseline show no differences (the slimmedCaloJets are not monitored)
- local tests show that slimmedCaloJets appear as expected. In ttbar events with PU35 (wf equivalent of 10224) on 3K events there are about 8 slimmedCaloJets per event and the increase in the file size is 240 bytes/event. 
    - The pending removal of caloJet userFloats would decrease the event size by 90 bytes/event, but it can happen only after updates in DQM. An issue #19605 was created to track this .

cmsbuild commented:
+1
Tested at: a933638f1bd3870b903b59318a273d00374a8722
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21237/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/6625d2da7769faa8b12de4b3703335f117f040fb
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21237/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21237/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21237/summary.html

Comparison Summary:
 - You potentially added 5 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1790655
 - DQMHistoTests: Total failures: 14587
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1775902
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

davidlange6 commented on PhysicsTools/PatAlgos/plugins/CaloJetSlimmer.cc:
```diff
@@ -0,0 +1,63 @@
+#include "CommonTools/UtilAlgos/interface/SingleElementCollectionSelector.h"
+
+#include "DataFormats/JetReco/interface/CaloJet.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
```
hi @jingyuluo - why is this dependency needed? 

Depending on the answer to that question, perhaps RecoJets will be a better location for this code.

jingyuluo replied:
@davidlange6 
Dear David, sorry it's not needed here, I'll remove the line here... Which subdirectory of RecoJets do you suggest to move this code to? (maybe JetAnalyzers?)

slava77 replied:
I think that it is better to hold this code here in this package, in view of possible evolution and also the scope of the workflows (miniAOD only).

jingyuluo committed:
remove the line for DataFormats/PatCandidates/interface/Jet.h

cmsbuild commented:
Pull request #19446 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 46778baa2be7938bf4769e34538e4124efda6b41


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/06199984952d9a6468923f7f1e47bed589272363
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21339/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21339/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21339/summary.html

I found follow errors while testing this PR

**Failed tests:  AddOn**

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun_MC.root : FAILED - time: date Mon Jul 10 23:23:36 2017-date Mon Jul 10 23:22:16 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-10-1100/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 23:23:36 2017-date Mon Jul 10 23:22:16 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 23:23:36 2017-date Mon Jul 10 23:22:16 2017 s - exit: 21248
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns --magField 38T_PostLS1 : FAILED - time: date Mon Jul 10 23:23:04 2017-date Mon Jul 10 23:22:21 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_PRef_MC.root : FAILED - time: date Mon Jul 10 23:23:25 2017-date Mon Jul 10 23:22:23 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-10-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 23:23:25 2017-date Mon Jul 10 23:22:23 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 23:23:25 2017-date Mon Jul 10 23:22:23 2017 s - exit: 21248
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 --magField 38T_PostLS1 : FAILED - time: date Mon Jul 10 23:23:54 2017-date Mon Jul 10 23:22:55 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun2016 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun2016_MC.root : FAILED - time: date Mon Jul 10 23:25:49 2017-date Mon Jul 10 23:23:06 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-10-1100/src/HLTrigger/Configuration/test/OnLine_HLT_GRun2016.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 23:25:49 2017-date Mon Jul 10 23:23:06 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun2016,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun2016 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun2016_MC.root --fileout file:RelVal_Raw_GRun2016_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 23:25:49 2017-date Mon Jul 10 23:23:06 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --magField 38T_PostLS1 --fileout file:RelVal_Raw_HIon_MC.root : FAILED - time: date Mon Jul 10 23:26:42 2017-date Mon Jul 10 23:24:02 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-10-1100/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 23:26:42 2017-date Mon Jul 10 23:24:02 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016,Run2_HI --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 23:26:42 2017-date Mon Jul 10 23:24:02 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_PIon_MC.root : FAILED - time: date Mon Jul 10 23:33:12 2017-date Mon Jul 10 23:30:10 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-10-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 23:33:12 2017-date Mon Jul 10 23:30:10 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 23:33:12 2017-date Mon Jul 10 23:30:10 2017 s - exit: 21248



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/06199984952d9a6468923f7f1e47bed589272363
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21339/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21339/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19446/21339/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1884334
 - DQMHistoTests: Total failures: 15044
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1869124
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/19446	46778ba
- there was only a technical update since the last signoff https://github.com/cms-sw/cmssw/pull/19446/files/a933638f1bd3870b903b59318a273d00374a8722..46778baa2be7938bf4769e34538e4124efda6b41
- jenkins tests pass

jingyuluo commented:
@davidlange6 @slava77 
 
Dear David and Slava,

 The test error seems to be unrelated to this PR... please let me know if there is any action item left for merging this PR.

Thanks, 
Jingyu

slava77 commented:
I have signed off already, two days ago.
@davidlange6 are there remaining issues with this PR?

slava77 commented:
@davidlange6 are there remaining issues with this PR?

davidlange6 commented:
i thought i had commented on this PR - is there a similar one around?

jingyuluo commented:
Hi @davidlange6 , I addressed the comment by removing the line "#include "DataFormats/PatCandidates/interface/Jet.h" ", but I'm not sure whether I should move the code to RecoJets? Which subdirectory of RecoJets do you suggest to move the code to?

Thanks,
Jingyu

davidlange6 commented:
ah - ok - I found my comment and your answer - was hidden by github.  I suspect this should be in RecoJets - but given the string cut parser dependence lets leave it here for now.

davidlange6 commented:
merge

arizzi commented:
do we have a 92X equivalent of this PR?

jingyuluo commented:
Hi @arizzi,  sorry there is no 92X equivalent of this PR... Should I create a 92X backport of this PR?

arizzi commented:
yes please!

jingyuluo commented:
Hi @arizzi , I just created a 92X backport here: #20041
